wire-ios-request-strategy/Sources/Request Strategies/Federation/TerminateFederationRequestStrategy.swift

```diff
@@ -0,0 +1,111 @@
+////
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+private let zmLog = ZMSLog(tag: "terminate federation")
+
+@objcMembers
+public final class TerminateFederationRequestStrategy: AbstractRequestStrategy {
+
+    // MARK: - Properties
+
+    private let entitySync: EntityActionSync
+
+    // MARK: - Life cycle
+
+    public override init(
+        withManagedObjectContext managedObjectContext: NSManagedObjectContext,
+        applicationStatus: ApplicationStatus
+    ) {
+        entitySync = EntityActionSync(actionHandlers: [])
+
+        super.init(
+            withManagedObjectContext: managedObjectContext,
+            applicationStatus: applicationStatus
+        )
+
+        configuration = [
+            .allowsRequestsWhileOnline,
+            .allowsRequestsDuringQuickSync,
+            .allowsRequestsWhileWaitingForWebsocket,
+            .allowsRequestsWhileInBackground
+        ]
+    }
+
+    // MARK: - Request
+
+    public override func nextRequestIfAllowed(for apiVersion: APIVersion) -> ZMTransportRequest? {
+        return entitySync.nextRequest(for: apiVersion)
+    }
+
+}
+
+// MARK: - Event processing
+
+extension TerminateFederationRequestStrategy: ZMEventConsumer {
+
+    public func processEvents(_ events: [ZMUpdateEvent],
+                              liveEvents: Bool,
+                              prefetchResult: ZMFetchRequestBatchResult?) {
```

Comment:
This is just my personal preference, but I think it's nicer when we put parameters on a new line because then the indentation is consistent and not dependent on the length of the method name:

```suggestion
    public func processEvents(
        _ events: [ZMUpdateEvent],
        liveEvents: Bool,
        prefetchResult: ZMFetchRequestBatchResult?
    ) {
```